src/figuresClasses.ts

```diff
@@ -1,19 +1,96 @@
 export interface Figure {
-
+  shape: string;
+  color: string;
+  getArea(): number;
 }
 
 export class Triangle implements Figure {
+  readonly shape: string;
+
+  readonly color: string;
+
+  readonly sideA: number;
+
+  readonly sideB: number;
+
+  readonly sideC: number;
 
+  constructor(color: string, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be greater than 0.');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('Not a triangle');
+    }
+
+    this.shape = 'triangle';
+    this.color = color;
+    this.sideA = a;
+    this.sideB = b;
+    this.sideC = c;
+  }
+
+  getArea(): number {
+    const s = (this.sideA + this.sideB + this.sideC) / 2;
+    const area = Math.sqrt(s * (s - this.sideA)
+    * (s - this.sideB) * (s - this.sideC));
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
 export class Circle implements Figure {
+  readonly shape: string;
+
+  readonly color: string;
 
+  readonly radius: number;
+
+  constructor(color: string, radius: number) {
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than 0.');
+    }
+
+    this.shape = 'circle';
+    this.color = color;
+    this.radius = radius;
+  }
+
+  getArea(): number {
+    const area = Math.PI * this.radius * this.radius;
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
 export class Rectangle implements Figure {
+  readonly shape: string;
+
+  readonly color: string;
+
+  readonly width: number;
+
+  readonly height: number;
+
+  constructor(color: string, width: number, height: number) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('Width and height must be greater than 0.');
+    }
+
+    this.shape = 'rectangle';
+    this.color = color;
+    this.width = width;
+    this.height = height;
+  }
```

Comment:
```suggestion
 shape: Shape = 'rectangle';

  constructor(public color: Color, private width: number, private height: number) {
    if (width <= 0 || height <= 0) {
      throw new Error('Width and height must be greater than 0');
    }
  }
```

As you remember you can set variables for constructor like this, this looks less complicated, do the same of all the similar cases above